allennlp/modules/text_field_embedders/basic_text_field_embedder.py

```diff
@@ -64,13 +64,28 @@ def get_output_dim(self) -> int:
         return output_dim
 
     def forward(self, text_field_input: Dict[str, torch.Tensor], num_wrapping_dims: int = 0) -> torch.Tensor:
-        if self._token_embedders.keys() != text_field_input.keys():
-            if not self._allow_unmatched_keys:
+        embedder_keys = self._token_embedders.keys()
+        input_keys = text_field_input.keys()
+
+        # Check for unmatched keys
+        if not self._allow_unmatched_keys:
+            if embedder_keys < input_keys:
+                # token embedder keys are a strict subset of text field input keys.
+                message = (f"Your text field is generating more keys ({list(input_keys)}) "
```

Comment:
Hmm I'm not sure this is the "most likely" error - perhaps we can modify this to say "If you are using a token embedder which requires multiple token indexers (such as the OpenaiEmbedder), you need to pass the `allow_unmatched_keys=True` flag to your BasicTextFieldEmbedder config. If not, you should check that there is a 1:1 mapping between your token indexers and token embedders."